backend/layers/business/business.py

```diff
@@ -147,6 +147,39 @@ def get_published_collection_version(self, collection_id: CollectionId) -> Optio
         """
         return self.database_provider.get_collection_mapped_version(collection_id)
 
+    def get_latest_published_collection_versions_by_schema(
+        self, schema_version: str
+    ) -> List[CollectionVersionWithPublishedDatasets]:
+        """
+        Returns a list with the latest published collection version that matches the given schema_version, for each
+        canonical collection
+        """
+        has_wildcards = "_" in schema_version
+        collection_versions = self.database_provider.get_collection_versions_by_schema(schema_version, has_wildcards)
+
+        # for each published canonical collection, map its most recently published collection version
+        collections = dict()
+        for collection_version in collection_versions:
+            if collection_version.published_at is None:
```

Comment:
you can filter these out in the database query.

Reply:
I know, though I got tripped up determining whether this is technically business logic. We generally seem to do timestamp processing in the business layer, though I can be convinced to switch it back

Reply:
Either way is fine by me.